### PR TITLE
EID-1243: Write acceptance tests for Error states within the proxy node.

### PR DESCRIPTION
**Pull request by tomrosier**
Added some new acceptance tests for different states in the proxy node.

Tidied up some of the test runner stuff to along the way.

Stub connector can now generate a bad AuthN request so we can test some of the validation.

EID-1258: Add landing page.
Added a landing page to the stub connector 

screen shot 2019-01-25 at 09 39 49

And added a button to the end of the journey to return to the start.

screen shot 2019-01-25 at 09 40 36

### CODE REVIEW

File: proxy-node-acceptance-tests/features/proxy_node.feature [hunk 1]
```diff
@@ -3,4 +3,21 @@ Feature: proxy-node feature
...
\ No newline at end of file
+        Then they should arrive at the success page
+
+    Scenario: IDP Generates Authn Failure
+        Given the user is at Stub Connector
+        And the stub idp supplies a "Authn Failure"
+        Then the user should be presented with an hub error page
+
+    Scenario: IDP Generates Authn Failure
+        Given the stub connector supply a bad authn request
```
Comment (dbes-gds) [inline proxy-node-acceptance-tests/features/proxy_node.feature]: typo: supplies

File: stub-connector/src/main/java/uk/gov/ida/notification/stubconnector/resources/SendAuthnRequestResource.java [hunk 2]
```diff
@@ -46,18 +46,44 @@ public SamlFormView setupAuthnRequest(@Session HttpSession session) throws Resol
                 AttributeConstants.EIDAS_DATE_OF_BIRTH_ATTRIBUTE_NAME
         );
 
-        String authnRequestId = UUID.randomUUID().toString();
+         return authnRequestGenerator.generate(
+                 UUID.randomUUID().toString(),
```
Comment (dbes-gds) [inline stub-connector/src/main/java/uk/gov/ida/notification/stubconnector/resources/SendAuthnRequestResource.java]: Would you mind putting an underscore at the start of the request id?  Had to do this to connect to the hub:

https://github.com/alphagov/verify-proxy-node/compare/EID-1223-connect-proxy-node-to-hub#diff-0c000cbcc20f5ae7883478b90896131f

**Comment by tomrosier** [comment]
Updated the logo on the start page.

screen shot 2019-01-25 at 10 08 01

**Comment by CharlesIC** [comment]
Has the new page been user researched and approved by the content designers? :trollface:

**Comment by JHjava** [comment]
@CharlesIC Don't think that is necessary for a stub

**Pull request merged**